File: Annexe 4/app/src/main/java/com/example/annexe4/MainActivity.java
package com.example.annexe4;

import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

import androidx.activity.EdgeToEdge;
import androidx.activity.result.ActivityResult;
import androidx.activity.result.ActivityResultCallback;
import androidx.activity.result.ActivityResultLauncher;
import androidx.activity.result.contract.ActivityResultContracts;
import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

import java.io.FileInputStream;
import java.io.FileOutputStream;
import java.io.ObjectInputStream;
import java.io.ObjectOutputStream;


public class MainActivity extends AppCompatActivity {

    TextView textView;
    Button bouton;
    ActivityResultLauncher<Intent> lanceur;
    Utilisateur utilisateur;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });

        Ecouteur ec = new Ecouteur();
        textView = findViewById(R.id.textView);
        bouton = findViewById(R.id.button);
        bouton.setOnClickListener(ec);

        lanceur = registerForActivityResult(new ActivityResultContracts.StartActivityForResult(), new CallBackConfirmer());
        UnSerialize();
        if (savedInstanceState != null){
            utilisateur = (Utilisateur)savedInstanceState.getSerializable("utilisateur");
            message();
        }

    }

    private void message(){
        String message = "Bonjour " + utilisateur.getPrenom() + " " + utilisateur.getNom() + " !";
        textView.setText(message);
    }

    private class CallBackConfirmer implements ActivityResultCallback<ActivityResult>{
        @Override
        public void onActivityResult(ActivityResult o) {
            if (o.getResultCode() == ConfirmerActivity.RESULT_OK){
                utilisateur = (Utilisateur) o.getData().getExtras().get("utilisateur");
                message();
            }
        }
    }

    private class Ecouteur implements View.OnClickListener{
        @Override
        public void onClick(View v) {
            if (v == bouton){
                lanceur.launch(new Intent(MainActivity.this, ConfirmerActivity.class));
                Serialize();
            }
        }
    }

    public void Serialize() {
        try {
            FileOutputStream fos = openFileOutput("fichier.ser", MODE_PRIVATE);
            ObjectOutputStream oos = new ObjectOutputStream(fos);
            oos.writeObject(utilisateur);
            oos.close();
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
    public void UnSerialize() {
        try {
            FileInputStream fis = openFileInput("fichier.ser");
            ObjectInputStream ois = new ObjectInputStream(fis);
            utilisateur = (Utilisateur) ois.readObject();
            message();

        } catch (Exception e) {
            e.printStackTrace();
        }
    }

    @Override
    protected void onStop() {
        super.onStop();
        Serialize();
    }

    @Override
    protected void onSaveInstanceState(@NonNull Bundle outState) {
        super.onSaveInstanceState(outState);
        outState.putSerializable("utilisateur", utilisateur);
    }
}

File: Annexe3C/projetDentiste/app/src/main/java/com/factures/applicationpaiementfactures/MainActivity.kt
package com.factures.applicationpaiementfactures

import android.os.Bundle
import android.widget.CheckBox
import android.widget.EditText
import android.widget.LinearLayout
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import java.io.FileInputStream
import java.io.FileOutputStream
import java.io.ObjectInputStream
import java.io.ObjectOutputStream

class MainActivity : AppCompatActivity() {

    lateinit var dent1: LinearLayout
    lateinit var dent2: LinearLayout
    var dent = Dent
    var dents: List<Dent> = emptyList() // Initialize as empty list

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContentView(R.layout.activity_main)
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }


        UnSerialize()

        dent1 = findViewById(R.id.dent1)
        dent2 = findViewById(R.id.dent2)

        if (dents.isNotEmpty()) {
            (dent1.getChildAt(0) as EditText).setText(dents[0].numDent)
            (dent1.getChildAt(1) as CheckBox).isChecked = dents[0].traitement
            (dent1.getChildAt(2) as EditText).setText(dents[0].notes)

            (dent2.getChildAt(0) as EditText).setText(dents[1].numDent)
            (dent2.getChildAt(1) as CheckBox).isChecked = dents[1].traitement
            (dent2.getChildAt(2) as EditText).setText(dents[1].notes)
        }

        val dentUn = Dent(
            (dent1.getChildAt(0) as EditText).text.toString(),
            (dent1.getChildAt(1) as CheckBox).isChecked,
            (dent1.getChildAt(2) as EditText).text.toString()
        )

        val dentDeux = Dent(
            (dent2.getChildAt(0) as EditText).text.toString(),
            (dent2.getChildAt(1) as CheckBox).isChecked,
            (dent2.getChildAt(2) as EditText).text.toString()
        )

        // Update dents list
        dents = listOf(dentUn, dentDeux)
    }

    override fun onStop() {
        super.onStop()
        Serialize()
    }

    public fun Serialize() {
        try {
            val fos: FileOutputStream = openFileOutput("fichier.ser", MODE_PRIVATE)
            val oos = ObjectOutputStream(fos)
            oos.writeObject(dents)
            oos.close()
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }

    public fun UnSerialize() {
        try {
            val fis: FileInputStream = openFileInput("fichier.ser")
            val ois = ObjectInputStream(fis)
            dents = ois.readObject() as List<Dent>
            ois.close()
        } catch (e: Exception) {
            e.printStackTrace()
            dents = emptyList()
        }
    }
}
